.github/workflows/setup-environment.yml

```diff
@@ -0,0 +1,25 @@
+name: setup-environment
+
+on:
+  workflow_call:
+
+env:
+  GO_VERSION: 1.23.8
```

Comment:
It'd be nice to define `GO_VERSION` somewhere centrally in our repo and have all workflows simply refer to a single source of truth, so any input on how to accomplish that in the cleanest way would be appreciated.